config/configmap/inferenceservice.yaml

```diff
@@ -546,7 +546,16 @@ data:
 
   deploy: |-
     {
-      "defaultDeploymentMode": "Serverless"
+      "defaultDeploymentMode": "Serverless",
+      "serviceAnnotationDisallowedList": [
```

Comment:
Let's separate out to a different config item, maybe we can call `labelAnnotationPropagation`